src/eq_schema/Question.js

```diff
@@ -19,12 +19,14 @@ class Question {
     const dateRange = findDateRange(question);
 
     if (dateRange) {
+      remove(question.answers, answer => {
+        return answer.type !== "DateRange";
+      });
       this.type = "DateRange";
-      this.answers = this.buildDateRangeAnswers(dateRange);
     } else {
       this.type = "General";
-      this.answers = this.buildAnswers(question.answers);
     }
+    this.answers = this.buildAnswers(question.answers);
```

Comment:
The fact that we have to add an `else if` to the `buildAnswers` function implies that it's not really doing what we want.

It might be worth creating a function that builds the composite's children instead? e.g. 

```js
if (dateRange) {
  const dateRangeAnswer = first(filter(question.answers, a => a.type === "DateRange"));
  this.answers = buildCompositeAnswers(dateRangeAnswer)
  this.type = "DateRange"
} else {
  this.answers = this.buildAnswers(question.answers);
  this.type = "General"
}